schema/xmlelement.cpp

```diff
@@ -104,4 +104,11 @@ Annotation::List XmlElement::annotations() const
 {
     return d->mAnnotations;
 }
+
+bool XmlElement::operator==(const XmlElement &other) const
+{
+    return name() == other.name();
```

Comment:
Surely this should compare nameSpace() too?

[and then you can remove the nameSpace comparison from subclasses that have it, like ComplexType]

Reply:
Just updated the code.
Still not sure if all == operator are complete. Should the isResolved be tested?
Should ComplexType also check for the derivedType and groups?

Reply:
isResolved() is not separate date, it's just the combination of type() and reference(), both of which you already test. So it makes no difference, it's just redundant. This is actually a good reason for writing `mType == other.mType` etc. instead of `type() == other.type()`. You'd easily spot what's not actual data members, in addition to avoiding all those method calls.

I think you can skip mDerivedTypes, this is bookkeeping that happens much later than the step where we decide to reuse an existing class.
Groups, however, are part of the complextype definition and should be tested.

Reply:
Anything left to do with the patch?

Reply:
Ah, I don't get notified when you just push more commits. Looks good now, let's merge this.